src/pages/question.js

```diff
@@ -0,0 +1,152 @@
+import { useEffect, useState } from "react";
+import NavBar from "../components/navBar";
+import Vector from "../assets/Vector.svg";
+import Enter from "../assets/Enter_icon.svg";
+import { useNavigate } from "react-router-dom";
+
+async function getAPIData(url) {
+  const response = await fetch(process.env.REACT_APP_API_URL + url, {
+    credentials: "same-origin",
+    headers: { Authorization: `Bearer ${localStorage.getItem("token")}` },
+  });
+  const json = await response.json();
+  return json;
+}
+
+async function post(url, body) {
+  const response = await fetch(process.env.REACT_APP_API_URL + url, {
+    method: "POST",
+    body: JSON.stringify(body),
+    headers: {
+      "Content-type": "application/json; charset=UTF-8",
+      Authorization: `Bearer ${localStorage.getItem("token")}`,
+    },
+  });
+  const json = await response.json();
+  return json;
+}
+
+const getQuestion = async () => {
+  var question_id = await getAPIData("weekly_question").then(
+    (response) => response["question_id"]
+  );
+  var content = await getAPIData("questions/" + question_id.toString()).then(
+    (response) => response["content"]
+  );
+  return content;
+};
+
+function Question() {
+  const [content, setContent] = useState("");
+  const [noAnswers, setNoAnswers] = useState(true);
```

Comment:
This works, but it's not necessary. If you want to know if there are no answers, just check if the length of `answers` is 0

Reply:
When data can be easily derived from existing state, it's better to avoid using a separate `useState` for it

Reply:
Ops, true 😅